server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -148,13 +150,52 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
      * Compression Level gives a trade-off between compression ratio and speed. The higher compression level results in higher compression ratio but slower compression and decompression speeds.
      * This setting is <b>not</b> realtime updateable.
      */
-    public static final Setting<Integer> INDEX_CODEC_COMPRESSION_LEVEL_SETTING = Setting.intSetting(
+
+    public static final Setting<Integer> INDEX_CODEC_COMPRESSION_LEVEL_SETTING = new Setting<>(
         "index.codec.compression_level",
-        3,
-        1,
-        6,
+        Integer.toString(3),
+        new Setting.IntegerParser(1, 6, "index.codec.compression_level", false),
         Property.IndexScope
-    );
+    ) {
+        @Override
+        public Set<SettingDependency> getSettingsDependencies(String key) {
+            return Set.of(new SettingDependency() {
+                @Override
+                public Setting<String> getSetting() {
+                    return INDEX_CODEC_SETTING;
+                }
+
+                @Override
+                public void validate(String key, Object value, Object dependency) {
+                    if (!(dependency instanceof String)) {
+                        throw new IllegalArgumentException("Codec should be of string type.");
+                    }
+                    doValidateCodecSettings((String) dependency);
+                }
+            });
+        }
+    };
+
+    private static void doValidateCodecSettings(final String codec) {
+        switch (codec) {
+            case "zstd":
+            case "zstd_no_dict":
+                return;
+            case "best_compression":
+            case "lucene_default":
+            case "default":
+                break;
+            default:
```

Comment:
Nit: Do we need this ```switch case```? doesn't the ```if condition``` below suffice?

The ```switch case``` does helps to optimize but the ```if condition``` makes it more generic.

Reply:
thanks @backslasht for the review!
Yes, we need the switch case because ```Codec.availableCodecs()``` returns the codecs registered with lucene. Here, for zstd/zstd_no_dict -> ```Lucene95CustomCodec``` is registered and for best_compression/default ```Lucene95Codec``` is registered. The switch helps us cover all the specific codecs, as well as optimize.